### PR TITLE
Update node install on build script

### PR DESCRIPTION
**Pull request by salazarm**
## Summary & Motivation
https://dagsterlabs.slack.com/archives/C03CCE471E0/p1696454877943329

## How I Tested These Changes
buildkite

### CODE REVIEW

File: .buildkite/dagster-buildkite/dagster_buildkite/steps/dagster_ui.py [hunk 1]
```diff
@@ -32,8 +32,9 @@ def build_dagster_ui_steps() -> List[CommandStep]:
             "cd js_modules/dagster-ui",
             # Explicitly install Node 16.x because BK is otherwise running 12.x.
             # Todo: Fix BK images to use newer Node versions, remove this.
```
Comment (hellendag) [inline .buildkite/dagster-buildkite/dagster_buildkite/steps/dagster_ui.py:34]: Update comment?

File: .buildkite/dagster-buildkite/dagster_buildkite/steps/dagster_ui.py [hunk 2]
```diff
@@ -32,8 +32,9 @@ def build_dagster_ui_steps() -> List[CommandStep]:
...
-            "curl -sL https://deb.nodesource.com/setup_16.x | bash -",
-            "apt-get -yqq --no-install-recommends install nodejs",
+            "sudo yum install"
+            " https://rpm.nodesource.com/pub_20.x/nodistro/repo/nodesource-release-nodistro-1.noarch.rpm -y",
+            "sudo yum install nodejs -y --setopt=nodesource-nodejs.module_hotfixes=1",
```
Comment (hellendag) [inline .buildkite/dagster-buildkite/dagster_buildkite/steps/dagster_ui.py:37]: What are these args based on?

Comment (salazarm) [reply .buildkite/dagster-buildkite/dagster_buildkite/steps/dagster_ui.py:37]: I copy pasted the recommended install command ¯\_(ツ)_/¯

File: js_modules/dagster-ui/packages/ui-core/src/asset-graph/sidebar/Sidebar.tsx [hunk 3]
```diff
@@ -47,7 +47,7 @@ export const AssetGraphExplorerSidebar = React.memo(
       : false;
 
     React.useEffect(() => {
-      if (selectWhenDataAvailable) {
+      if (selectWhenDataAvailable && selectedNodeHasDataAvailable) {
```
Comment (gibsondan) [inline js_modules/dagster-ui/packages/ui-core/src/asset-graph/sidebar/Sidebar.tsx:50]: unrelated i assume, remove?

**Review by gibsondan — CHANGES_REQUESTED** [review]
looks like buildkite doesn't have yum sadly

**Comment by gibsondan** [comment]
(based on BK results)

**Pull request closed without merging**